src/coreclr/vm/i386/jitinterfacex86.cpp

```diff
@@ -442,6 +442,22 @@ void *JIT_TrialAlloc::GenBox(Flags flags)
     // jne              noAlloc
     sl.X86EmitCondJump(noAlloc, X86CondCode::kJNE);
 
+    // Save the MethodTable ptr
+    sl.X86EmitPushReg(kECX);
```

Comment:
Would it be better to introduce a new JIT box helper and return it from `getBoxHelper` instead of slowing down and complicating the super fast helpers?

Reply:
This was indeed annoying. Lots more changes, but it did help me to find something I missed in NativeAOT.